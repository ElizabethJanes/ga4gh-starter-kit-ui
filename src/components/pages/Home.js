import '@fontsource/roboto';
import React from 'react';
import Typography from '@material-ui/core/Typography';
import {
  Link
} from "react-router-dom";
import Button from '@material-ui/core/Button';
import { Container } from '@material-ui/core';

const Home = () => {
    return(
      <div align="center">
          <meta
            name="viewport"
            content="minimum-scale=1, initial-scale=1, width=device-width"
        />
        <Container maxWidth="lg">
          <Typography variant="h1" gutterBottom>Welcome to the GA4GH Starter Kit</Typography>
          <Typography variant="h3" gutterBottom>Get Started</Typography>
          <Typography variant="body1" gutterBottom>Click the buttons below to start using one of the GA4GH Starter Kits</Typography>
            <div>
              <nav>
                <Button variant="outlined" color="primary" size="large">
                  <Link to='/drs'>DRS Starter Kit</Link>
                </Button>
              </nav>
            </div>
        </Container>
      </div>
    );
  }

  export default Home;