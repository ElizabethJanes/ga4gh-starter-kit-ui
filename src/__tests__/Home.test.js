import React from 'react';
import renderer, {act} from 'react-test-renderer';
import Home from '../lib/components/pages/Home';
import { unmountComponentAtNode } from "react-dom";
import {MemoryRouter} from 'react-router-dom'

let container = null;
beforeEach(() => {
    container = document.createElement("div");
    document.body.appendChild(container);
});

afterEach(() => {
    unmountComponentAtNode(container);
    container.remove();
    container = null;
});

test('<Home />', () => {
    let home = renderer
    .create(
        <MemoryRouter>
            <Home/>
        </MemoryRouter>
    )
    .toJSON();
    expect(home).toMatchSnapshot();
})