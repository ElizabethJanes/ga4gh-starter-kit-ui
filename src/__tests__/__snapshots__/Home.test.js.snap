// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> 1`] = `
<div
  align="center"
>
  <meta
    content="minimum-scale=1, initial-scale=1, width=device-width"
    name="viewport"
  />
  <div
    className="MuiContainer-root MuiContainer-maxWidthLg"
  >
    <h1
      className="MuiTypography-root MuiTypography-h1 MuiTypography-gutterBottom"
    >
      Welcome to the GA4GH Starter Kit
    </h1>
    <h3
      className="MuiTypography-root MuiTypography-h3 MuiTypography-gutterBottom"
    >
      Get Started
    </h3>
    <p
      className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
    >
      Click the buttons below to start using one of the GA4GH Starter Kits
    </p>
    <div>
      <nav>
        <a
          aria-disabled={false}
          className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge"
          href="/drs"
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex={0}
        >
          <span
            className="MuiButton-label"
          >
            <span
              className="MuiTypography-root MuiTypography-button"
            >
              DRS Starter Kit
            </span>
          </span>
        </a>
      </nav>
    </div>
  </div>
</div>
`;
